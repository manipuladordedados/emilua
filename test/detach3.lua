-- This test ensures that fibers whose joinable fiber handles got collected
-- will have their stacktrace printed to stderr (if err'ed)

local println = require('println')
local sleep_for = require('sleep_for')

collectgarbage("stop")

local fib = spawn(function()
    println('secondary fiber starts')
    error('tag', 0)
end)

println('secondary fiber spawned')
sleep_for(10)
println('end of main fiber')
