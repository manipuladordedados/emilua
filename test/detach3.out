secondary fiber spawned
secondary fiber starts
end of main fiber
Fiber 0x1 panicked: 'tag'
stack traceback:
	input.lua:8: in function <input.lua:6>
	[C]: in function ''
	[string "?"]: in function <[string "?"]:0>
